src/Microsoft.DotNet.Build.Tasks/FilterRuntimesFromSupports.cs

```diff
@@ -0,0 +1,294 @@
+                    }
+                    for (int i = 0, k = 0; i < toRemove.Count; i++)
+                    {
+                        if (!toRemove[i])
+                        {
+                            compatibilityProfile.RestoreContexts.RemoveAt(k);
+                        }
+                        else
+                        {
+                            k++;
+                        }
+                    }
+                    filteredRuntimes.Add(compatibilityProfile);
+                }
+            }
+            else
+            {
+                filteredRuntimes = filteredSupports;
+            }
+
+            JObject projectJson = ReadProject(InputProjectJson);
+            JObject supportsClause = new JObject();
+            foreach (var compatibilityProfile in filteredRuntimes)
+            {
+                Dictionary<string, List<string>> map = new Dictionary<string, List<string>>();
+                foreach (var tfmRidPair in compatibilityProfile.RestoreContexts)
+                {
+                    if (map.ContainsKey(tfmRidPair.Framework.GetShortFolderName()))
+                    {
+                        List<string> str = new List<string>();
+                        map.TryGetValue(tfmRidPair.Framework.GetShortFolderName(), out str);
+                        str.Add(tfmRidPair.RuntimeIdentifier);
+                    }
+                    else
+                    {
+                        map.Add(tfmRidPair.Framework.GetShortFolderName(), new List<string>() { tfmRidPair.RuntimeIdentifier });
+                    }
+                }
+                var tfmRidsForSupports = new JObject();
+                foreach (var item in map)
+                {
+                    tfmRidsForSupports[item.Key] = new JArray(item.Value);
+                }
+                supportsClause[compatibilityProfile.Name] = tfmRidsForSupports;
+            }
+
+            if (projectJson["supports"]!=null)
+            {
+                projectJson["supports"] = supportsClause;
+            }
+
+            string projectJsonRoot = JsonConvert.SerializeObject(projectJson, Formatting.Indented) + Environment.NewLine;
+
+            if (!projectJson.Equals(File.ReadAllText(InputProjectJson)))
+            {
+                Directory.CreateDirectory(Path.GetDirectoryName(InputProjectJson));
+                File.WriteAllText(InputProjectJson, projectJsonRoot);
+            }
+
+            return true;
+        }
+
+        private static JObject ReadProject(string projectJsonPath)
+        {
+            using (TextReader projectFileReader = File.OpenText(projectJsonPath))
+            {
+                var projectJsonReader = new JsonTextReader(projectFileReader);
+                var serializer = new JsonSerializer();
+                return serializer.Deserialize<JObject>(projectJsonReader);
+            }
+        }
+
+
+        public static RuntimeGraph ReadRuntimeGraph(string filePath)
```

Comment:
Why didn't you use [JsonRuntimeFormat.ReadRuntimeGraph](https://github.com/NuGet/NuGet.Client/blob/88330551e465e5673505c525515e74129f73d98c/src/NuGet.Core/NuGet.RuntimeModel/JsonRuntimeFormat.cs#L17)?